src/FuelClient.cc

```diff
@@ -572,12 +572,16 @@ Result FuelClient::DownloadModel(const ModelIdentifier &_id,
   if(!res)
     return res;
 
-  for (auto dep : dependencies)
+  for (ModelIdentifier dep : dependencies)
   {
-    auto dep_res = this->DownloadModel(dep, _headers);
+    // Download dependency if not in the local cache
+    if (!this->dataPtr->cache->MatchingModel(dep))
+    {
+      auto dep_res = this->DownloadModel(dep, _headers);
```

Comment:
Style nit (I know it's pre-existing though)

```suggestion
      auto depRes = this->DownloadModel(dep, _headers);
```

Reply:
Changed in [352f42d](https://github.com/gazebosim/gz-fuel-tools/pull/263/commits/352f42d3ea70743024145517f37203f23b68259f)